lib/sender.js

```diff
@@ -80,12 +80,24 @@ class Sender {
       }
 
       skipMasking = (mask[0] | mask[1] | mask[2] | mask[3]) === 0;
-      if (options.readOnly && !skipMasking) merge = true;
 
       offset = 6;
     }
 
-    let payloadLength = data.length;
+    let dataLength;
+    if (typeof data === 'string') {
+      if (!options.mask || skipMasking) {
+        dataLength = Buffer.byteLength(data);
+      } else {
+        data = toBuffer(data);
+        dataLength = data.length;
+      }
+    } else {
+      dataLength = data.length;
```

Comment:
`data` can be a `DataView` an `ArrayBuffer`, etc. Writing them to a socket is not supported.

Reply:
If `data` is not a string then it's expected to be a buffer, just like before.

Reply:
i.e. nothing has changed in terms of behaviour other than we check for string type.

Reply:
Maybe I am misreading but this change

```js
const buf = perMessageDeflate ? toBuffer(data) : data;
```

in `send()` keep data as is if permessage-deflate is disabled.

Reply:
You were right. Fixed.